### ISSUE TITLE
Deprecated actions checkConfirmation / checkCreate are being used

### ISSUE THREAD

**Issue by sypets** [in2code-de/powermail#1101]
_Updated text (for correctness, sypets)._

-----

Confirmation.html:

```
<f:form
	action="checkCreate"
```

https://github.com/in2code-de/powermail/blob/master/Resources/Private/Templates/Form/Confirmation.html#L43

FormController.php

```
public function checkCreateAction(Mail $mail): ResponseInterface
    {
        trigger_error(
            'EXT:powermail -- Method "checkCreateAction" is deprecated since version 12.3.2, will be removed in version 13.0.0',
            E_USER_DEPRECATED
        );

```

This is also the case for checkCreate.

So, this results in unnecessary deprecation log.

Also, this results in problems in extensions which rely on the name of the actions, for example

* powermailrecaptcha
* ~~powermailcaptcha~~ (already fixed)

**powermailrecaptcha:**

```
 protected function isCaptchaCheckToSkip(): bool
    {
        if (property_exists($this, 'flexForm')) {
            $confirmationActive = $this->flexForm['settings']['flexform']['main']['confirmation'] === '1';
            return $this->getActionName() === 'create' && $confirmationActive;
        }
        return false;
    }

```
https://github.com/in2code-de/powermailrecaptcha/blob/master/Classes/Domain/Validator/SpamShield/RecaptchaMethod.php#L110

powermailcaptcha (already fixed):

```
 protected function isCaptchaCheckToSkip(): bool
    {

...
            if (($action === 'create' || $action === 'checkCreate') && $confirmationActive) {
```

https://github.com/thomasrawiel/powermailcaptcha/blob/develop/Classes/Domain/Validator/SpamShield/CaptchaMethod.php#L187

## Reproduce

1. Use a captcha extension, e.g. powermailcaptcha with friendly captcha
2. Create form with confirmation page
3. Add a captcha field to the form
4. (optional) activate spam log
5. Fill out the form and solve the captcha
6. submit the form

The form will not be submitted. Possibly flash message displayed. You can see in Spam log this is reported as spam.

Confirm that in the URL you see the check action, e.g. `https://mysite.org/form?tx_powermail_pi1%5Baction%5D=checkConfirmation&tx_powermail_pi1%5Bcontroller%5D=Form"`

## Version

* TYPO3 12.4.18
* powermail: 12.3.5
* PHP 8.2

**Comment by sypets** [comment]
Additional note: I did not notice this before, specifically the problem with the catpcha extension (we use a modified powermailcaptcha) because we had overridden `Confirmation.html`, but this resulted in other problems after upgrade to v12, so we used the original again.

Then, the captcha no longer worked with forms with confirm page.

I had to add the "checkCreate" action to the check in my patched powermailcaptcha:

I am a bit puzzled, no one reported this. :confused:

**Comment by mschwemer** [comment]
Thanks.

Background for the deprecation is, that I consider the `check*` actions as an antipattern. 

We introduced them with the TYPO3 v12 compatible version, because we did not know it better back then. 

I still do not have a perfect solution for TYPO3 v13 comaptibility, but I want to get rid of it. The deprecation notice is there, because there will be a breaking change for v13, in case somebody extended / xclassed it.
